src/hotspot/share/opto/predicates.cpp

```diff
@@ -381,6 +382,83 @@ bool TemplateAssertionPredicateExpressionNode::is_template_assertion_predicate(N
   return node->is_If() && node->in(1)->is_Opaque4();
 }
 
+InitializedAssertionPredicate::InitializedAssertionPredicate(IfNode* template_assertion_predicate, Node* new_init,
+                                                             Node* new_stride, PhaseIdealLoop* phase)
+    : _template_assertion_predicate(template_assertion_predicate),
+      NOT_PRODUCT(_assertion_predicate_type(template_assertion_predicate->assertion_predicate_type()) COMMA)
+      _new_init(new_init),
+      _new_stride(new_stride),
+      _phase(phase) {}
+
+// Create an Initialized Assertion Predicate at the provided control from the _template_assertion_predicate:
+//
+//          init                 stride
+//            |                    |
+//   OpaqueLoopInitNode  OpaqueLoopStrideNode                      _new_init    _new_stride
+//                  \     /                                               \     /
+//                    ...                                                   ...
+//                     |                                                     |
+//                    Bool                                                new Bool
+//                     |                                                     |
+//                  Opaque4                ======>              control   OpaqueInitializedAssertionPredicate
+//                     |                                              \  /
+//                    If                                             new If
+//                  /    \                                           /    \
+//            success     fail path                         new success   new Halt
+//              proj    (Halt or UCT)                           proj
+//
+IfTrueNode* InitializedAssertionPredicate::create(Node* control) {
+  IdealLoopTree* loop = _phase->get_loop(control);
+  OpaqueInitializedAssertionPredicateNode* new_opaque_bool = create_new_bool(control);
```

Comment:
This is confusing: I ask for a `bool` but get an `opaque`...

Reply:
Better might be: `create_initialized_assertion_predicate_expression`

Reply:
See comment below.